File: data/src/main/java/com/icehousecorp/maunorafiq/data/forecast/repository/datasource/ForecastDataStoreFactory.java
package com.icehousecorp.maunorafiq.data.forecast.repository.datasource;

import javax.inject.Inject;
import javax.inject.Singleton;

import retrofit2.Retrofit;

/**
 * Created by maunorafiq on 11/29/16.
 */
@Singleton
public class ForecastDataStoreFactory {

    private Retrofit retrofit;

    @Inject
    public ForecastDataStoreFactory(Retrofit retrofit) {
        this.retrofit = retrofit;
    }

    public ForecastDataStore create() {
        ForecastDataStore forecastDataStore;

        forecastDataStore = new CloudForecastDataStore(retrofit);

        return forecastDataStore;
    }
}

File: data/src/main/java/com/icehousecorp/maunorafiq/data/forecast/repository/datasource/ForecastDataStoreFactory.kt
package com.icehousecorp.maunorafiq.data.forecast.repository.datasource

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by maunorafiq on 11/29/16.
 */
@Singleton
class ForecastDataStoreFactory
@Inject
constructor() {

    fun create(): ForecastDataStore {
        val forecastDataStore: ForecastDataStore

        forecastDataStore = CloudForecastDataStore()

        return forecastDataStore
    }
}
